batch_job/src/batch_job_mesos.c

```diff
@@ -0,0 +1,437 @@
+static pid_t mesos_scheduler_pid = 0;
+
+static void start_mesos_scheduler(struct batch_queue *q)
+{
+
+	pid_t mesos_pid;
+	mesos_pid = fork();			
+	mesos_scheduler_pid = mesos_pid;
+	char *mesos_cwd;
+	mesos_cwd = path_getcwd();
+
+	char exe_path[MAX_BUF_SIZE];
+
+	if(readlink("/proc/self/exe", exe_path, MAX_BUF_SIZE) == -1) {
+		debug(D_ERROR, "read \"proc/self/exe\" fail\n");
+		exit(EXIT_FAILURE);
+	}
+
+	char exe_dir_path[MAX_BUF_SIZE];
+	path_dirname(exe_path, exe_dir_path);
+
+    char *exe_py_path = string_format("%s/mf_mesos_scheduler", exe_dir_path);
+	char *ld_preload_str = NULL;
+
+	if(mesos_preload != NULL) {
+		ld_preload_str = string_format("LD_PRELOAD=%s", mesos_preload);
+	} 
+	char *envs[] = {ld_preload_str, NULL};
+
+	char *mesos_python_path = xxstrdup(mesos_py_path);
+
+	if (mesos_python_path == NULL) {
+		debug(D_ERROR, "Please specify the mesos python path\n");
+		exit(EXIT_FAILURE);
+	}
+
+	if (mesos_pid > 0) {
+
+		debug(D_INFO, "Start makeflow mesos scheduler.");
+
+	} else if (mesos_pid == 0) {
+
+		const char *batch_log_name = batch_queue_supports_feature(q, "batch_log_name");
+
+		int mesos_fd = open(batch_log_name, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+		if (mesos_fd == -1) {
+	    	debug(D_ERROR, "Failed to open %s \n", batch_log_name);
+	    	exit(EXIT_FAILURE);
+	    }
+
+	    if (dup2(mesos_fd, 1) == -1) {
+	    	debug(D_ERROR, "%s\n", strerror(errno));
+	    	exit(EXIT_FAILURE);
+	   	}
+
+		if (dup2(mesos_fd, 2) == -1) {
+	    	debug(D_ERROR, "%s\n", strerror(errno));
+	    	exit(EXIT_FAILURE);
+	   	}
+
+	    close(mesos_fd);
+
+		execle("/usr/bin/python", "python", exe_py_path, mesos_cwd, mesos_master, mesos_python_path, (char *) 0, envs);
+
+		_exit(127);
+
+	} else {
+
+		debug(D_ERROR, "mesos batch system couldn't create new process: %s\n", strerror(errno));
+		exit(EXIT_FAILURE);
+
+	}
+
+}
+
+
+static batch_job_id_t batch_job_mesos_submit (struct batch_queue *q, const char *cmd, \
+		const char *extra_input_files, const char *extra_output_files, \
+		struct jx *envlist, const struct rmsummary *resources )
+{
+
+	// Get the path to mesos python site-packages
+	if (is_mesos_py_path_known != 1) {
+		mesos_py_path = batch_queue_get_option(q, "mesos-path");
+		if (mesos_py_path == NULL) {
+			debug(D_ERROR, "Please set the path to mesos python library\n");
+			exit(EXIT_FAILURE);
+		} else {
+			debug(D_INFO, "Get mesos_path %s from command line\n", mesos_py_path);
+			is_mesos_py_path_known = 1;
+		}
+	}
+
+	// Get the mesos master address
+	if (is_mesos_master_known != 1) {
+		mesos_master = batch_queue_get_option(q, "mesos-master");
+		if (mesos_master == NULL) {
+			debug(D_ERROR, "Please specify the ip address or domain name of mesos master\n");
+			exit(EXIT_FAILURE);
+		} else {
+			debug(D_INFO, "Get mesos_path %s from command line\n", mesos_py_path);
+			is_mesos_master_known = 1;
+		}
+	}
+
+	mesos_preload = batch_queue_get_option(q, "mesos-preload");
+
+	if (is_mesos_py_path_known == 1 && \
```

Comment:
Do not use \.

Better:  if(is_mesos_py...   && is_mesos_master ...  && !is_scheduler_running) {